src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/finance/TransactionEventListener.kt

```diff
@@ -36,16 +37,25 @@ class TransactionEventListener(
           if (eventFeatureSwitch.isEnabled(eventType, "transactions")) {
             val messageId = UUID.fromString(sqsMessage.MessageId)
             when (eventType) {
-              "OFFENDER_TRANSACTIONS-INSERTED" -> transactionSynchronisationService.transactionInserted(
+              "OFFENDER_TRANSACTIONS-INSERTED" -> transactionSynchronisationService.transactionInsertCheck(
                 sqsMessage.Message.fromJson(),
                 messageId,
               )
               // NOTE: Several inserts per second!
+              /*
+              when event arrives, check DB table of txn_ids:
+               if not there, add and requeue message for 5s time
+               if it is there and target time not yet arrived (or not expired), ignore
+              when requeued message arrives: remove from cache if necessary and call api
+               */
```

Comment:
Actually this comment isnt entirely accurate now Im checing it - there is no stored 'target time'